packages/analytics/amplify_analytics_pinpoint/example/integration_test/identify_user_test.dart

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+//  http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+import 'dart:async';
+
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'utils/mock_data.dart';
+import 'utils/setup_utils.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('recordEvent', () {
+    late final Stream<Map<String, Object?>> eventsStream;
+
+    setUpAll(() async {
+      eventsStream = await configureAnalytics();
+    });
+
+    tearDownAll(Amplify.reset);
+
+    // TODO(fjnoyp): Running this locally too many times will cause max endpoint error of 15 from Pinpoint
+    // We need to delete/reset Analytics backend somehow
```

Comment:
Is it because you're using the same userId below? The pinpoint backend doesn't need to be reset each time, you just need to present as a different user, either by changing `userId` or making sure Auth is signedOut on teardowns.

Reply:
thanks! That seems to fix it by changing userId.